low-level/files/opt/seagate/sspl/prerequisites/sspl_dev_deploy.sh

```diff
@@ -0,0 +1,484 @@
+        echo "ERROR: Consul is not available. \
+        For consul and other 3rd party package installation, \
+        check prereq script usage. Exiting."  2>&1 | tee -a "${LOG_FILE}"
+        exit 1
+    fi
+    # Create config file for consul agent
+    CONSUL_CONFIG_DIR='/opt/seagate/cortx/sspl/bin'
+    CONSUL_CONFIG_FILE="$CONSUL_CONFIG_DIR/consul_config.json"
+    if [ ! -d "$CONSUL_CONFIG_DIR" ]; then
+        mkdir -p "${CONSUL_CONFIG_DIR}"
+    fi
+    rm -rf $CONSUL_CONFIG_FILE
+    cat << EOF >> $CONSUL_CONFIG_FILE
+        {
+            "watches": [
+                {
+                    "type": "key",
+                    "key": "sspl/config/SYSTEM_INFORMATION/log_level",
+                    "args": ["/opt/seagate/cortx/sspl/bin/consume_cfg_change_alert"]
+                }
+            ]
+        }
+EOF
+    # Invoke consul agent
+    CONSUL_PS=$(pgrep "consul" || true)
+    [ -z "$CONSUL_PS" ] &&
+        consul agent -dev -config-file=$CONSUL_CONFIG_FILE &>/dev/null &
+
+    TRIES=0
+    while [ -z "$CONSUL_PS" ]; do
+        sleep 3
+        TRIES=$((TRIES+1))
+        CONSUL_PS=$(pgrep "consul" || true)
+        if [ $TRIES -gt 10 ]; then
+            echo "ERROR: Consul service is not started"  2>&1 | tee -a "${LOG_FILE}"
+            break
+        fi
+    done
+    echo "Done consul setup."  2>&1 | tee -a "${LOG_FILE}"
+
+}
+
+
+insert_config_in_consul(){
+
+    # Install configobj
+    configobj_req="$(grep configobj ${SSPL_BASE_DIR}/low-level/requirements.txt)"
+    [ -n "$configobj_req" ] && python3 -m pip install $configobj_req
+
+    # Update persistent cluster_id
+    consul_cluster_id="$(consul kv get system_information/cluster_id ||:)"
+    config_cluster_id="$(sed -nr 's/^cluster_id=([^,]+)$/\1/p' $CONFIG_FILE | head -1)"
+    if [ -n "$consul_cluster_id" ]; then
+        cluster_id=$consul_cluster_id
+    elif [ -n "$config_cluster_id" ]; then
+        cluster_id=$config_cluster_id
+    else
+        cluster_id=$(uuidgen)
+    fi
+    consul kv put system_information/cluster_id "$cluster_id"
+    echo "cluster_id: $cluster_id"
+
+    # Update persistent node_id
+    consul_node_id="$(consul kv get system_information/${NODE}/node_id ||:)"
+    config_node_id="$(sed -nr 's/^node_id=([^,]+)$/\1/p' $CONFIG_FILE | head -1)"
+    if [ -n "$consul_node_id" ]; then
+        node_id=$consul_node_id
+    elif [ -n "$config_node_id" ]; then
+        node_id=$config_node_id
+    else
+        node_id=$(uuidgen)
+    fi
+    consul kv put system_information/${NODE}/node_id "$node_id"
+    echo "node_id: $node_id"
+
+    # Encrypt passwords
+    if [ -n "$RMQ_PASSWD" ]; then
+        rbmq_key=$(python3 -c 'from cortx.utils.security.cipher import Cipher; \
+            print(Cipher.generate_key('"'$cluster_id'"', "rabbitmq"))')
+        RMQ_PASSWD=$(python3 -c 'from cortx.utils.security.cipher import Cipher; \
+            print(Cipher.encrypt('$rbmq_key', '"'$RMQ_PASSWD'"'.encode()).decode("utf-8"))')
```

Comment:
dont print passwords. keep the line commented, we will enable only during development

Reply:
OKay.